packages/plugin-js-packages/src/lib/runner/outdated/transform.ts

```diff
@@ -0,0 +1,102 @@
+import { Issue } from '@code-pushup/models';
+import { objectToEntries, pluralize } from '@code-pushup/utils';
+import { PackageDependency } from '../../config';
+import { outdatedSeverity } from './constants';
+import {
+  NormalizedOutdatedEntries,
+  NormalizedVersionOverview,
+  NpmOutdatedResultJson,
+  PackageVersion,
+  VersionType,
+} from './types';
+
+export function outdatedResultToAuditOutput(
+  result: NpmOutdatedResultJson,
+  dependenciesType: PackageDependency,
+) {
+  // current might be missing in some cases
+  // https://stackoverflow.com/questions/42267101/npm-outdated-command-shows-missing-in-current-version
+  const validDependencies: NormalizedOutdatedEntries = objectToEntries(result)
+    .filter(
+      (entry): entry is [string, NormalizedVersionOverview] =>
+        entry[1].current != null,
+    )
+    .filter(([, detail]) =>
+      dependenciesType === 'prod'
+        ? detail.type === 'dependencies'
+        : detail.type.startsWith(dependenciesType),
+    );
+  const outdatedDependencies = validDependencies.filter(
+    ([, versions]) => versions.current !== versions.wanted,
+  );
+
+  const issues =
+    outdatedDependencies.length === 0
+      ? []
+      : outdatedToIssues(outdatedDependencies);
+
+  return {
+    slug: `npm-outdated-${dependenciesType}`,
+    score: outdatedDependencies.length === 0 ? 1 : 0,
+    value: outdatedDependencies.length,
```

Comment:
A really wouldn't go with a binary score here. There will be always be some possible patch or minor update available and that's fine. I suggest only using major versions to calculate the score and/or making it relative to the total number of dependencies.

For reference, this is how I mapped it in a customer demo:

```ts
        const filteredDeps = deps.filter(dep => dep.pkgType === pkgType)
        const issues = deps.filter(dep => dep.pkgType === pkgType).flatMap(depToIssues)

        const depsCount = filteredDeps.length
        const majorIssuesCount = issues.filter(issue => issue.severity === "error").length

        return {
          slug,
          score: (depsCount - majorIssuesCount) / depsCount,
          value: majorIssuesCount,
          displayValue: `${majorIssuesCount} out of ${depsCount} dependencies require major update`,
          details: { issues },
        }

```

Reply:
If it's out of outdated, then I would probably format it differently so it's clear, e.g. _10 outdated dependencies, 3 requiring major update_

Also added an alternative suggestion here: https://github.com/code-pushup/cli/pull/567#discussion_r1528846912